packages/block-library/src/column/edit.native.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect } from '@wordpress/data';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import {
+	InnerBlocks,
+	BlockControls,
+	BlockVerticalAlignmentToolbar,
+} from '@wordpress/block-editor';
+import { withViewportMatch } from '@wordpress/viewport';
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+
+function ColumnEdit( {
+	attributes,
+	setAttributes,
+	hasChildren,
+	isSelected,
+	getStylesFromColorScheme,
+	isParentSelected,
+	isDescendantOfParentSelected,
+	columnsContainerSettings,
+	isMobile,
+} ) {
+	const { verticalAlignment } = attributes;
+	const {
+		columnsInRow = 1,
+		width: columnsContainerWidth,
+	} = columnsContainerSettings;
+
+	const columnContainerBaseWidth = styles[ 'column-container-base' ].maxWidth;
+	const containerMaxWidth = styles[ 'columns-container' ].maxWidth;
+
+	const containerWidth = columnsContainerWidth || containerMaxWidth;
+
+	const minWidth = Math.min( containerWidth, columnContainerBaseWidth );
+	const columnBaseWidth = minWidth / columnsInRow;
+
+	const applyBlockStyle = ( placeholder = false ) => {
+		if ( isMobile ) {
+			return;
+		}
+
+		const pullWidths = ( names ) =>
+			names.map(
+				( name ) =>
+					( styles[ `column-${ name }-margin` ] || {} ).width || 0
+			);
+
+		let width = columnBaseWidth;
+		const names = [
+			'selected',
+			'parent-selected',
+			'descendant-selected',
+			'placeholder-selected',
+		];
+		const widths = pullWidths( names );
+		const [ , parentSelected, , placeholderSelected ] = widths;
+
+		if ( isParentSelected ) {
+			width -= placeholder ? placeholderSelected : parentSelected;
+			return { width };
+		}
+
+		const [ selected, , descendantSelected ] = widths;
+
+		if ( placeholder ) {
+			if ( isDescendantOfParentSelected ) {
+				width -= selected;
+			} else {
+				width -=
+					columnsInRow === 1 ? parentSelected : placeholderSelected;
+				if ( ! hasChildren ) width -= 4;
+			}
+		} else if ( isSelected ) {
+			width -= ! hasChildren ? selected : descendantSelected;
+		} else if ( isDescendantOfParentSelected ) {
```

Comment:
> DescendantOfParent
Does it correspond to sibling?

Reply:
It's applied to siblings of the block as well. However the scope is wider because it applies to all blocks that has parent included in the parent tree of selected block